akka-http-core/src/test/scala/akka/http/scaladsl/SslConfigWarningsSpec.scala

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import akka.actor.ActorSystem
+import akka.event.Logging
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.testkit.{ EventFilter, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.{ Matchers, WordSpec }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class SslConfigWarningsSpec extends WordSpec with Matchers {
+  val testConf: Config = ConfigFactory.parseString("""
+    akka.loglevel = INFO
+    akka.stdout-loglevel = INFO
+    akka.log-dead-letters = OFF
+    akka.http.server.request-timeout = infinite
+
+    akka.ssl-config {
+      loose {
+        disableSNI = true
+        disableHostnameVerification = true
+      }
+    }
+    """)
+
+  "ssl-config loose options should cause warnings to be logged" should {
+
+    "warn if SNI is disabled globally" in {
+      implicit val system = ActorSystem(getClass.getSimpleName, testConf)
+      implicit val materializer = ActorMaterializer()
+
+      val p = TestProbe()
+      system.eventStream.subscribe(p.ref, classOf[Logging.LogEvent])
+
+      EventFilter.warning(start = "Detected that Server Name Indication (SNI) is disabled globally", occurrences = 1) intercept {
+        Http()(system)
+      }
+
+      // the very big warning shall be logged only once per actor system (extension)
+      EventFilter.warning(start = "Detected that Server Name Indication (SNI) is disabled globally", occurrences = 0) intercept {
+        Http()(system)
```

Comment:
Yes and this one is specifically added and much louder, discussed it a bit on gitter, story here is:

- ssl-config logs a warning when created and bad things noticed, it's a small oneliner (bad, once per ssl-config)
- akka logs a big fat warning when ssl-config is GLOBALLY (very bad) configured to disable things, instead we recommend setting it up for a single connection 
  - note that if you do so, you'll still get one warning message that for that ssl-config instance (the above "bad" triggers then and logs once).

This is to not hide that you're using loose settings somewhere. Worst case is the global one - 2 log messages will be shown, but the akka one specifically telling about "hey, do this per connection plz"